client/src/auth/sign-up.tsx

```diff
@@ -117,106 +106,106 @@ export default function SignUpApp() {
+  try {
+    userDetails = JSON.parse(searchParams.get("user_details") || "{}");
+  } catch (jsonParseError) {
+    console.warn("The 'user_details' was not valid JSON");
+  }
+
   return (
-    <form
-      method="post"
-      onSubmit={(event) => {
-        event.preventDefault();
-        if (checkedTerms) {
-          submitSignUp();
-        }
-      }}
-    >
+    <>
+      {userDetails.avatar_url && (
+        <img
+          src={userDetails.avatar_url}
+          className="avatar"
+          alt="User profile avatar_url"
+        />
+      )}
+
+      {provider && (
+        <DisplaySignupDetails
+          provider={provider}
+          username={userDetails.name || ""}
+        />
+      )}
+
       {signupError && (
-        <div className="notecard error">
-          <h4>Signup Error</h4>
+        <div className="notecard negative">
           <p>
-            <code>{signupError.toString()}</code>
+            <strong>Signup error</strong> <code>{signupError.toString()}</code>
           </p>
         </div>
       )}
 
-      <DisplaySignupProvider provider={provider || ""} />
-      <DisplayUserDetails details={searchParams.get("user_details") || ""} />
-
-      <label htmlFor="id_terms">
-        <input
-          id="id_terms"
-          type="checkbox"
-          name="terms"
-          checked={checkedTerms}
-          onChange={(event) => {
-            setCheckedTerms(event.target.checked);
-          }}
-        />{" "}
-        I agree to Mozilla's{" "}
-        <a
-          href="https://www.mozilla.org/about/legal/terms/mozilla"
-          target="_blank"
-          rel="noreferrer noopener"
-        >
-          Terms
-        </a>{" "}
-        and{" "}
-        <a
-          href="https://www.mozilla.org/privacy/websites/"
-          target="_blank"
-          rel="noreferrer noopener"
-        >
-          Privacy Notice
-        </a>
-        .
-      </label>
+      <form
+        className="complete-sign-in"
+        method="post"
+        onSubmit={(event) => {
+          event.preventDefault();
+          if (checkedTerms) {
+            submitSignUp();
+          }
+        }}
+      >
+        <label htmlFor="id_terms">
+          <input
+            id="id_terms"
+            type="checkbox"
+            name="terms"
+            required
+            checked={checkedTerms}
+            onChange={(event) => {
+              setCheckedTerms(event.target.checked);
+            }}
+          />{" "}
+          I agree to Mozilla's{" "}
+          <a
+            href="https://www.mozilla.org/about/legal/terms/mozilla"
+            target="_blank"
+            rel="noreferrer noopener"
+          >
+            Terms
+          </a>{" "}
+          and{" "}
+          <a
+            href="https://www.mozilla.org/privacy/websites/"
+            target="_blank"
+            rel="noreferrer noopener"
+          >
+            Privacy Notice
+          </a>
+          .
+        </label>
```

Comment:
I don't love the fact that it's a label. Because labels are meant to be clicked on to check an adjacent checkbox. 
But I think it's still bad that the label wraps onto two lines where there's plenty of space on either side. 
<img width="647" alt="Screen Shot 2021-03-18 at 1 35 56 PM" src="https://user-images.githubusercontent.com/26739/111671952-d70fc300-87ef-11eb-9cf9-484d6a102eee.png">

Reply:
> I don't love the fact that it's a label. Because labels are meant to be clicked on to check an adjacent checkbox.

You can totally do that.

> But I think it's still bad that the label wraps onto two lines where there's plenty of space on either side.

I am going to respectfully disagree here. I feel it is much neater this way and keeps a nice rhythmic flow as mentioned previously.

Reply:
But it looks broken. Why is "Privacy Notice" on its own line? Why can't "Terms" be on its own line?
It looks like the whole label is situated inside a container that has a max-width that forces it to wrap, which is jarring because there's plenty of whitespace on either horizontal side.